### PR TITLE
Task 2 scoring

### PR DESCRIPTION
**Pull request by vpchung**
The scoring script is done, and is now ready for you to integrate into the workflow, @mdsage1! I listed some action items below to help walk you through it. Lmk if you'd like to jump on a call and do a working session as well.

Feel free to continue working off of this branch.

---

## Changelog
* add scoring script where metric(s) calculated are:
   - accuracy (@chepyle please let us know if another metric should be added) 
* ...

## Remaining TODOs
- [x] update the Dockerfile to include the new scoring script.  You may also need to include a new dependency on L5.
- [x] build the latest Docker image and push it up to [staging](https://www.synapse.org/#!Synapse:syn51703346), using the same Docker image name and tag version.
- [x] update `score.cwl` so that:
    - [x] the inline Python script is removed
    - [x] `dockerPull` utilizes our scoring model (the one pushed to staging)
    - [x] the list of `arguments` prefixes matches those in the Python scoring script, e.g. `-f` is an unknown CLI option
- [x] after updating the CWL script, try running it with `cwl-runner` to ensure it works as expected, e.g.
   ```
   cwl-runner steps/score.cwl --goldstandard path/to/gold.tsv --input_file path/to/pred.tsv
   ```
   > **Note**: if you would like to check whether scoring is working, you can generate a random sample using the goldstandard file:
   > ```python
   > >>> import pandas as pd
   > >>> import numpy as np
   > >>> res = pd.read_table('gold.tsv', usecols={'Participant_ID': np.int64, 'Disease_Name': str})
   > >>> res['Disease_Name'] = res.Disease_Name.sample(frac=1).values
   > >>> res.to_csv('pred.tsv', sep="\t", index=False)
   > ```
- [x] update the `score` step of `task2-workflow.cwl` as needed (from a quick glance, I don't think anything needs to be done, but let's double-check just in case!)

### CODE REVIEW

**Comment by mdsage1** [comment]
The score.py script is trying to use sklearn which has been deprecated and replaced w/scikit-learn. Consequently, you cannot install sklearn. Is this where L5 comes in? @vpchung

**Comment by vpchung** [comment]
@mdsage1 installing scikit-learn is perfect!  [`sklearn.metrics`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html) is available from that :-)

**Comment by vpchung** [comment]
@mdsage1 thanks again for working on this!  I did a quick review, and there are a couple of things that we'll need to address.

For the most part though, it looks great! 🚀

**Comment by vpchung** [comment]
Thanks, @mdsage1 - great work!

> Checked the score step of `task2-workflow.cwl` for needed updates (none identified)

The list of CWL inputs has been updated, e.g. `input_file` → `predictions_file`, so these params will also need to be updated in the main workflow file as well.  For example, `task2_workflow.cwl` is using the param name, [`inputs_file`, to pass along the predictions filepath to `score.cwl`](https://github.com/Sage-Bionetworks-Challenges/RARE-X-DREAM-Challenge-Infra/blob/82bfda83733934711436c7086ed1acbcc285049d/task2_workflow.cwl#L278). See comment above for more context.

File: steps/score.cwl [hunk 1]
```diff
@@ -6,34 +6,12 @@ label: Score predictions file
...
-                  'submission_status': prediction_file_status}
-        with open(args.results, 'w') as o:
-          o.write(json.dumps(result))
 
 inputs:
-  - id: input_file
+  - id: predictions_file
     type: File
-  - id: goldstandard
+  - id: goldstandard_file
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply steps/score.cwl:13]: I've removed the change in score.cwl and left task2_workflow unchanged. The id is now goldstandard for both.

File: steps/score.cwl [hunk 2]
```diff
@@ -6,34 +6,12 @@ label: Score predictions file
...
-        prediction_file_status = "SCORED"
-
-        result = {'auc': score,
-                  'submission_status': prediction_file_status}
-        with open(args.results, 'w') as o:
-          o.write(json.dumps(result))
 
 inputs:
-  - id: input_file
+  - id: predictions_file
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply steps/score.cwl:11]: task2_workflow.cwl has been updated to reflect the id for score as predictions_file, which is used in score.cwl and score.py

File: steps/score.cwl [hunk 3]
```diff
@@ -6,34 +6,12 @@ label: Score predictions file
...
     type: File
-  - id: goldstandard
+  - id: goldstandard_file
     type: File
-  - id: check_validation_finished
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply steps/score.cwl]: I have added the validation check back into the cwl file.

**Comment by mdsage1** [comment]
@vpchung I've implemented the changes you suggested. Please let me know if I've missed or seem to have misunderstood anything. Thanks!

**Review by vpchung — COMMENTED** [review]
@mdsage1 I'm not able to approve the changes (since I opened the PR), but LGTM!  We should be good to merge now 🎉 

One more thing -
I think this warning is safe to ignore, but just in case, do you mind running the scoring model on the instance to ensure it can run okay?  I get the following warning on my machine:

```
$ docker run --rm  docker.synapse.org/syn51198356/evaluation:v4 validate.py
WARNING: The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64) and no specific platform was requested
```

**Comment by mdsage1** [comment]
Ran:
```
$ docker run --rm  docker.synapse.org/syn51198356/evaluation:v4 validate.py
WARNING: The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64) and no specific platform was requested
 ```
The instance is running and will merge these branches.
